### ISSUE TITLE
CA1052 doesn't factor in derivation

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#4968]
### Analyzer

**Diagnostic ID**: [CA1052](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1052): `Static holder types should be Static or NotInheritable`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: 5.0.3 (Latest)

### Describe the bug

CA1052 doesn't factor in that types may be derived from the target type, in which case marking it static is erroneous.

### Steps To Reproduce

1. Enable CA1052.
2. Set its visibility to target internal and private: `dotnet_code_quality.CA1052.api_surface = private, internal`
3. Paste this into the app:
```C#
internal class Base
{
    public static int Value = 42;
}

internal class Derived : Base
{
    public Derived() => Console.WriteLine(Value);
}
```

### Expected behavior

No warning.  Even though Base on its own could be marked static, the presence of Derived means you can no longer mark Base as static.

### Actual behavior

Warns that Base should be marked static.

**Comment by Youssef1313** [comment]
Related to https://github.com/dotnet/roslyn-analyzers/issues/1820

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/1820
